File: 0401_0500/0446/0446.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n, m = (int(x) for x in f.readline().split())
    a = "YES"
    s = [4 if c == 'R' else 2 if c == 'G' else 1 if c == 'B' else 0 for c in ''.join(f.readline().strip() for i in range(n))]
    p = [int(x) for x in f.read().split()]
    for i in range(m * n):
        if s[i] and not s[i] & p[i]:
            a = "NO"
            break
    q.write(a)

File: 0401_0500/0446/0446.c
#include <stdio.h>
#include <ctype.h>
#include <stdlib.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n, m, r = 1;
	char *s, c;

	fscanf(f, "%d%d", &n, &m);
	s = (char*)malloc(n * m);
	for (int i = 0; i < n * m; ++i)
	{
		while (isspace(c = getc(f)));
		s[i] = (c == 'R' ? 4 : c == 'G' ? 2 : c == 'B' ? 1 : 0);
	}
	for (int i = 0, d; r && i < n * m; ++i)
	{
		fscanf(f, "%d", &d);
		if (s[i] && !(s[i] & d))
			r = 0;
	}
	fprintf(q, r ? "YES" : "NO");
	return 0;
}
